Mathlib/Combinatorics/SimpleGraph/Sum.lean

```diff
@@ -73,4 +74,60 @@ def Embedding.sumInr : H ↪g G ⊕g H where
   inj' u v := by simp
   map_rel_iff' := by simp
 
+/-- Color `G ⊕g H` with colorings of `G` and `H` -/
+def Coloring.sum (cG : G.Coloring γ) (cH : H.Coloring γ) : (G ⊕g H).Coloring γ := Coloring.mk
+  (Sum.elim cG cH) <| by
+  intro u v
+  cases u <;> cases v <;> simp
+  · exact cG.valid
+  · exact cH.valid
+
+/-- Get coloring of `G` from coloring of `G ⊕g H` -/
+def Coloring.sum_left (c : (G ⊕g H).Coloring γ) : G.Coloring γ := Coloring.mk (c ∘ Sum.inl) <| by
+  intro u v h
+  exact c.valid h
+
+/-- Get coloring of `H` from coloring of `G ⊕g H` -/
+def Coloring.sum_right (c : (G ⊕g H).Coloring γ) : H.Coloring γ := Coloring.mk (c ∘ Sum.inr) <| by
+  intro u v h
+  exact c.valid h
+
+/-- Color `G ⊕g H` with `Fin (n + m)` given a coloring of `G` with `Fin n` and a coloring of `H`
+with `Fin m` -/
+def Coloring.sum_fin {n m : ℕ} (cG : G.Coloring (Fin n)) (cH : H.Coloring (Fin m)) :
+    (G ⊕g H).Coloring (Fin (max n m)) := Coloring.mk
+  (Sum.elim (Fin.castMax m ∘ cG) (Fin.castMax' n ∘ cH)) <| by
+  intro u v
+  cases u <;> cases v <;> rename_i u v <;> simp <;> rw [Fin.ext_iff]
+  · rw [Fin.castMax_val, Fin.castMax_val, ← Fin.ext_iff]
+    exact cG.valid
+  · rw [Fin.castMax'_val, Fin.castMax'_val, ← Fin.ext_iff]
+    exact cH.valid
+
+theorem Colorable.sum_fin {n m : ℕ} (hG : G.Colorable n) (hH : H.Colorable m) :
+    (G ⊕g H).Colorable (max n m) := Nonempty.intro (hG.some.sum_fin hH.some)
+
+theorem Colorable.sum_left {n : ℕ} (h : (G ⊕g H).Colorable n) : G.Colorable n :=
+  Nonempty.intro (h.some.sum_left)
+
+theorem Colorable.sum_right {n : ℕ} (h : (G ⊕g H).Colorable n) : H.Colorable n :=
+  Nonempty.intro (h.some.sum_right)
+
+theorem ChromaticNumber.left_le_sum : G.chromaticNumber ≤ (G ⊕g H).chromaticNumber := by
```

Comment:
I failed to find any lemmas that use `SimpleGraph.ChromaticNumber` namespace. IMHO, we should just have `chromaticNumber_sum`.